core/src/main/java/org/elasticsearch/common/settings/AbstractScopedSettings.java

```diff
@@ -162,6 +180,37 @@ public synchronized Settings applySettings(Settings newSettings) {
     }
 
     /**
+     * Validates that all settings in the builder are registered and valid
+     */
+    public final void validate(Settings.Builder settingsBuilder) {
+        for (Map.Entry<String, String> entry : settingsBuilder.internalMap().entrySet()) {
+            validate(entry.getKey(), entry.getValue());
+        }
+    }
+
+    /**
+     * * Validates that all given settings are registered and valid
+     */
+    public final void validate(Settings settings) {
+        for (Map.Entry<String, String> entry : settings.getAsMap().entrySet()) {
+            validate(entry.getKey(), entry.getValue());
+        }
+    }
+
+
+    /**
+     * Validates that the setting is valid
+     */
+    public final void validate(String key, String value) {
+        Settings.Builder builder = Settings.builder().put(key, value);
+        Setting setting = get(key);
+        if (setting == null) {
+            throw new IllegalArgumentException("unknown setting [" + key + "]");
+        }
+        setting.get(builder.build());
```

Comment:
This seems like a pretty heavy way to validate a setting. Like, its building a bunch of maps and things.